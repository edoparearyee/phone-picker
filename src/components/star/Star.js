import React, { Component } from 'react';
import './Star.css';

class Star extends Component {
  render() {
    return (
      <svg
        className={`Star ${
          this.props.active ? 'Star--active' : 'Star--inactive'
        }`}
      >
        <polygon
          points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
          style={{ fillRule: 'nonzero' }}
        />
      </svg>
    );
  }
}

export default Star;
