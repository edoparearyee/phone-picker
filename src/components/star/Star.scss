@import "../../styles/variables";
@import "../../styles/functions";

.Star {
  height: rem(25px);
  width: rem(23px);
}

.Star--inactive {
  fill: $star-color-inactive;
}

.Star--active {
  fill: $star-color-active;
}
