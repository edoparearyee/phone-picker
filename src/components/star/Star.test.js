import React from 'react';
import ReactDOM from 'react-dom';
import Star from './Star';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Star />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('Star has class Star-inactive', () => {
  const component = renderer.create(<Star active={false} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Star has class Star-active', () => {
  const component = renderer.create(<Star active={true} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
