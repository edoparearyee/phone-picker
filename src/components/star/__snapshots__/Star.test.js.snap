// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Star has class Star-active 1`] = `
<svg
  className="Star Star--active"
>
  <polygon
    points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
    style={
      Object {
        "fillRule": "nonzero",
      }
    }
  />
</svg>
`;

exports[`Star has class Star-inactive 1`] = `
<svg
  className="Star Star--inactive"
>
  <polygon
    points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
    style={
      Object {
        "fillRule": "nonzero",
      }
    }
  />
</svg>
`;
